src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs

```diff
@@ -318,22 +322,31 @@ private void OutputToOutputWindow(string message)
                 }
             }
 
-            foreach (var reference in commandLineArguments.MetadataReferences)
+            foreach (var reference in commandLineArguments.ResolveMetadataReferences(project.CurrentCompilationOptions.MetadataReferenceResolver))
```

Comment:
Once we call `projectContext.SetOptions()` on line 269, the compilationOptions will be set, and the metadatareference resolver can't be null.